### ISSUE TITLE
re-saving multiple decks to same deck different w/version

### ISSUE THREAD

**Issue by caldwelljt2** [Epix37/Hearthstone-Deck-Tracker#777]
Ok, so this is a bit complex because the tracker gets it right the first time but wrong the second time. This isn't urgent, because the work around is easy, but maybe it could be fixed for future versions.

I had a lot of decks named with -mod# appended prior to deck versioning being added. I decided to go back and collapse them into deck versions and started opening each -mod# and removing the -mod# from the name and selecting the corresponding version number. It worked as expected for -mod1, but -mod2 and up attempted to save as version 1.1 even though a 1.1 already existed. (and it did successfully do it, meaning I had multiple version 1.1s of the same deck.).

If that's hard to follow, here is the raw names, what I changed the name to, etc.

1. HANDLOCK > Unchanged
2. HANDLOCK-mod1 > resaved as HANDLOCK (and selected v1.1 from the drop down)
3. HANDLOCK-mod2 > resaved as HANDLOCK (still only had v1.1 or 2.0 available, so had to select 1.1)

At this point I just re-edit the 2nd one and saved again to get prompted to go to v1.2, but I would think it should prompt me on #3 to save as version 1.2 since 1.1 existed.

Hope this helps, no rush on a fix, thanks for all the hard work guys!

p.s. let me know if i'm just doing something wrong, i.e. if the intention is to just rev to 2.0 an on. I've assumed v2.0 was something I would reserve for large scale changes to a deck (not one or two cards). Like if I wanted to create a totally different handlock based on different combos, etc.

**Comment by dvide** [comment]
I think you're confusing how the deck versioning works a little, and the problem is that there is no way at present to turn an existing deck into a version of another deck.

The way you're supposed to use deck versions is by using the version selector at the top right. So if you have 5 versions of the same deck, there would only be one deck shown in the deck list and 5 versions of that deck available to choose from in the combo box here:
![untitled](https://cloud.githubusercontent.com/assets/180188/6597470/dcb11f8c-c7f4-11e4-9618-a9238cac216c.jpg)

If I were to edit this deck, I'd have the option of saving my changes as v1.5 (overwriting the latest version), or v1.6, or v2.0. Any of these would still only show one `NO POWER HERE` deck in the deck list, but if I'd saved as v1.6 I could then choose that from the version selector.

When you resaved `HANDLOCK-mod1` as `HANDLOCK`, that just meant that you ended up with two decks both called `HANDLOCK`. It basically just renamed `HANDLOCK-mod1` as `HANDLOCK`, but it didn't overwrite the existing deck called `HANDLOCK` because you are allowed to have two decks with the same name.

So you had two separate decks, both called `HANDLOCK`, with neither having any versions. Then you saved v1.1 of the deck you had just renamed. This just created a separate version of that renamed deck, identical to the first version. What I mean is, HearthstoneDeckTracker doesn't know that the two decks, both called `HANDLOCK`, are supposed to be versions of the same deck. It thinks they're two separate decks, and one of them now has two versions (v1.0 and v1.1).

Likewise, when you saved `HANDLOCK-mod2` as `HANDLOCK` v1.1, that again simply renamed `HANDLOCK-mod2` to `HANDLOCK` and created a v1.1. That's correct though because, again, `HANDLOCK-mod2` was considered a separate deck with no versions before, so the next version up is v1.1 (or v2.0).

Unfortunately there is currently no support for consolidating existing decks together though, such that they would all appear like my screenshot as multiple versions of one deck. 

In the next release I believe you'll be able to do it manually though, if you want to take the time. In the next release, you'll be able to move multiple games at once from one deck to another deck, and [choose a specific version](https://github.com/Epix37/Hearthstone-Deck-Tracker/issues/691#issuecomment-77688834) that you want to assign the games to. So when the next release hits, I believe you could recreate all of your versions off of the original `HANDLOCK` deck, then move all the games assigned to the 'mod' decks to their appropriate versions.

To be more clear, what I mean is to do this (only when the next release hits). Rename your modified decks back to `HANDLOCK-mod1`, `HANDLOCK-mod2`, etc. (so that they're easier to see which one is which). Then go to your original `HANDLOCK` and edit the deck. Make the changes that would be required to bring it in line with `HANDLOCK-mod1` and then save it as v1.1. Then edit the `HANDLOCK` deck again and make the changes that would be required to bring it in line with `HANDLOCK-mod2`, and save the deck as v1.2. Keep doing this until you've recreated all the versions that you have. You should now be able to select which version you want from the combobox on `HANDLOCK`, just like my screenshot, but the problem is with the stats. The games that were played on `HANDLOCK-mod1` are still assigned to that deck.

But then you'd go into your game stats, and move all games assigned to `HANDLOCK-mod1` to `HANDLOCK` (v1.1), all games assigned to `HANDLOCK-mod2` to `HANDLOCK` (v1.2), etc. Then I believe when you've done that for all the decks, you can delete the `HANDLOCK-mod1`, `HANDLOCK-mod2`, decks etc. The next release will support moving multiple games at the same time, and moving the games to a specific version of a deck.

Hope this all makes sense. There's a lot in my post and it's complicated to explain so I apologize.

**Comment by GiggliG** [comment]
The way i solved this was as follows:

* change the name of each deck you want to combine, like Handlock1, Handlock2, etc., so they sort in the deck list view properly
* export screenshots of each deck you want to combine into one
* Create a new deck, call it Handlock-new, set it up with the cards from the handlock1, and save it 
* select handlock1, look at stats view, move all games into handlock-new.
* now go back to handlock-new, edit it, and enter the deck used in handlock2. Save it, and select version 1.1.
* select handlock2, look at stats view, move all games to handlock-new, making sure version 1.1 is selected.
* repeat until all handlock1, handlock2, etc. decks have been combined into handlock-new
* review the handlock-new deck, make sure everything is correct
* now delete all the old decks, and rename handlock-new to Handlock.

That's pretty tedious, but once it's done, it's done. It's very easy to proceed from there, making new versions when you need to and HDT will handle the stat tracking properly from now on.
